File: js/siebog/agents/MobileAgent1.js
importScripts("/siebog-war/js/radigost/agent.js");

function MobileAgent1() {		
	this.num = 42;
};

MobileAgent1.prototype = new Agent();

MobileAgent1.prototype.onInit = function(args, _post, _log, _moveToServer) {

};

MobileAgent1.prototype.onArrived = function(host, isServer) {		
	this.num++;
	if(isServer === true){
		this.isServer = true;
		print("PRINT: I'm at " +host +" server, and the number is " +this.num);
		console.log("CONSOLE.LOG: I'm at " +host +" server, and the number is " +this.num);
	}else{
		this.isServer = false;
		console.log("I just came from " +host +" to my browser, and the number is " +this.num);
	}
};

MobileAgent1.prototype.onMessage = function(msg) {
	if (typeof msg == "string") {
		msg = JSON.parse(msg);
	}			
	var content = msg.content;
	if(this.isServer === true){
		print("I'm on the server and I received the following message:\n" +msg);
		print("Message content: " +msg.content);
		if(content === "move"){
			print("Got the message to move back to the client.");
			this.moveToClient();
		}else{
			this.moveToServer(msg.content);
		}
	}else{
		console.log("A'm in a client browser, and I have received the following message: " +msg); 
		if(content === "move"){
			this.moveToServer();
		}
	}
};


/*
 * This links this agent to its Worker. 
 */
setAgentInstance(new MobileAgent());

File: java/siebog/agents/MobileAgent1.java
package siebog.agents;

import siebog.agentmanager.XjafAgent;
import siebog.agentmanager.Agent;
import siebog.messagemanager.ACLMessage;
//import siebog.agents.Agent;
import dnars.siebog.DNarsAgent;
import siebog.agents.AgentBuilder;
//import siebog.agents.AgentInitArgs;
import siebog.interaction.ACLMessage;
import siebog.SiebogClient;
import javax.ejb.Remote;
import javax.ejb.Stateless;

@Stateless
@Remote(Agent.class)
public class MobileAgent1 extends XjafAgent {
	int num = 42;

	@Override
	protected void onInit(AgentInitArgs args){
		
	}

	@Override
	protected void onMessage(ACLMessage msg) {				
		String content = msg.content;
		if(this.isServer == true){
			System.out.println("I'm on the server and I received the following message:\n" +msg);
			System.out.println("Message content: " +msg.content);
			if(content == "move"){
				System.out.println("Got the message to move back to the client.");
				this.move();
			}else{
				this.move(msg.content);
			}
		}else{
			LoggerUtil.log("A'm in a client browser, and I have received the following message: " +msg);
			if(content == "move"){
				this.move();
			}
		}
		
		if (msg.performative == Performative.RESUME) {	//Arrived
			this.num++;
			if(isServer == true){
				this.isServer = true;
				System.out.println("PRINT: I'm at " +host +" server, and the number is " +this.num);
				LoggerUtil.log("CONSOLE.LOG: I'm at " +host +" server, and the number is " +this.num);
			}else{
				this.isServer = false;
				LoggerUtil.log("I just came from " +host +" to my browser, and the number is " +this.num);
			}
		}
	}	

	@Override
	protected void doTasks(AgentInitArgs args) {
		
	}

	/*
	public static void main(String[] args) {
		SiebogClient.connect("localhost");
		AgentBuilder.module("dnars-web").ejb(MobileAgent1.class).randomName().start();
	} */
}